app/src/main/java/com/ably/tracking/demo/publisher/ably/log/LogFileWriterImpl.kt

```diff
@@ -0,0 +1,25 @@
+package com.ably.tracking.demo.publisher.ably.log
+
+import java.io.FileWriter
+
+class LogFileWriterImpl(private val fileManager: FileManagerImpl) : LogFileWriter {
+
+    private var fileWriter: FileWriter? = null
+
+    override val isReady: Boolean
+        get() = fileWriter != null
+
+    override fun prepare(directory: String, fileName: String) {
+        val file = fileManager.createFile(directory, fileName)
+        fileWriter = FileWriter(file)
+    }
+
+    override fun writeLine(line: String) {
+        fileWriter!!.appendLine(line)
+        fileWriter!!.flush()
```

Comment:
It is possible to call those functions in any order, and in such a case, this line will throw a `NullPointerException.`
I decided to avoid wrapping those in custom exceptions (in a few other places too) for simplicity of the demo apps, do you think it is worth changing that?

Reply:
Agreed. This can be accepted for demo apps.